[bmanara] iP

bmanara opened the pull request:
## Llama 
Llama helps with keeping track of your tasks! It supports:
1. Adding new tasks
2. Marking and unmarking tasks
3. Deletion of tasks
4. Different tasks types (eg. Todo, Deadline, Event)

Llama is,
- Simple to use
- Text-based (for you programming geeks)
- _SUPER FAST_ to use
- AND **FREE** :shipit:

Features (Implemented and Upcoming):
- [X] Managing Todos
- [X] Managing Deadlines
- [X] Managing Events
- [ ] Reminders (coming soon!)

> “Every great developer you know got there by solving problems they were unqualified to solve until they actually did it.” – Patrick McKenzie, software engineer ([source])(https://www.codecademy.com/resources/blog/inspirational-coding-quotes/)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application. launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

bmanara committed:
Rename Duke to Llama

bmanara committed:
Add exit message

bmanara committed:
Add "bye" command

bmanara committed:
Add echo in Llama class

bmanara committed:
Add new feature to add tasks

bmanara committed:
Add displayString static method

bmanara committed:
Change taskList to taskArray

bmanara committed:
Add 'list' command

bmanara committed:
Add Task Class

bmanara committed:
Override toString method in Task Class

bmanara committed:
Add markDone and markNotDone methods in Task

bmanara committed:
Add feature to mark and unmark tasks

bmanara committed:
Add response to mark and unmark

bmanara committed:
Add Todo Class

bmanara committed:
Add 'Todo' command

bmanara committed:
Remove taskArray, add TaskStorage class

bmanara committed:
Add Deadline Task

bmanara committed:
Add Event Task

bmanara committed:
Update test files

bmanara committed:
Add total number of tasks for response

bmanara committed:
Add feature for bot the deal with empty task input

bmanara committed:
Update test files to test error handling

bmanara committed:
Add wrong input handling for tasks and unknown commands

bmanara committed:
Add handling of invalid input for unmark and mark tasks

bmanara committed:
Change from taskArray to taskList

bmanara committed:
Add 'delete' command

bmanara committed:
Update test files for delete command

bmanara committed:
Implement file creation for data saving

bmanara committed:
Implement directory creation

bmanara committed:
Implement task saving when adding task

bmanara committed:
Implement loading tasks into TaskStorage

bmanara committed:
Implement deleting task from file

bmanara committed:
Implement updating tasks when marked or unmarked

bmanara committed:
Merge branch 'branch-Level-7'

bmanara committed:
Add support for dates

bmanara committed:
Add support for saving time

bmanara committed:
Merge branch 'branch-Level-8'

bmanara committed:
Change constant variable naming

bmanara committed:
Add Ui class

bmanara committed:
Reformat Llama.java

bmanara committed:
Add TaskList class

bmanara committed:
Add Parser and AddCommand class

bmanara committed:
Add ExitCommand

bmanara committed:
Add ListCommand

bmanara committed:
Add EditCommand

bmanara committed:
Add DeleteCommand

bmanara committed:
Merge branch 'branch-A-MoreOOP'

bmanara committed:
Move all files into packages

bmanara committed:
Merge branch 'branch-A-Packages'

bmanara committed:
Merge branch 'add-gradle-support'

bmanara committed:
Add gradle

bmanara committed:
Add test cases for Tasks

bmanara committed:
Add tests for AddCommand

bmanara committed:
Modify build.gradle

bmanara committed:
Add JavaDoc to commands

bmanara committed:
Add JavaDoc to data

bmanara committed:
Add JavaDoc to exceptions

bmanara committed:
Add JavaDoc to parser

bmanara committed:
Add JavaDoc to ui

bmanara committed:
Add JavaDoc to Llama

bmanara committed:
Update code to follow coding standards

bmanara committed:
Add find command

bmanara committed:
Add JavaDoc for new methods

bmanara committed:
Merge branch 'branch-A-JavaDoc'

bmanara committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/data/Storage.java

bmanara committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/data/TaskList.java

bmanara committed:
Implement checkStyle

bmanara committed:
Change code to follow style

bmanara committed:
Merge branch 'branch-A-CheckStyle'

juancarlovieri commented on src/main/java/Llama.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    /**
+     * Method to start running the program
```
Shouldn't the first sentence of a method's JavaDoc start in the form of a verb? (According to [this](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=In%20method%20header%20comments%2C%20the%20first%20sentence%20should%20start%20in%20the%20form%20Returns%20...%2C%20Sends%20...%2C%20Adds%20...%20etc.%20(not%20Return%20or%20Returning%20etc.)))

I also noticed this in other JavaDoc comments.

juancarlovieri commented on src/main/java/data/Event.java:
```diff
@@ -0,0 +1,39 @@
+
+    @Override
+    public String getSaveTaskString() {
+        return "E" + super.getSaveTaskString() + "|" +
+                this.startTime.format(FORMATTER) + "|" + this.endTime.format(FORMATTER);
```
```suggestion
        return "E" + super.getSaveTaskString() + "|"
                + this.startTime.format(FORMATTER) + "|" + this.endTime.format(FORMATTER);
```

I think operators should be in a new line when there is a line wrap.

juancarlovieri commented on src/main/java/data/TaskList.java:
```diff
@@ -0,0 +1,153 @@
+    public void searchTasks(String searchStr, Ui ui) {
+        int numberOfTasks = taskList.size();
+        int num = 1;
+        ui.displayString("Here are the tasks with keyword '" + searchStr +"':");
```
```suggestion
        ui.displayString("Here are the tasks with keyword '" + searchStr + "':");
```

I think operators must be followed by whitespace.

juancarlovieri commented on src/main/java/exceptions/InvalidTaskException.java:
```diff
@@ -0,0 +1,10 @@
+ * Represents an invalid task input exception
+ */
+public class InvalidTaskException extends Exception {
+    public InvalidTaskException(String message) {
```
I think there is a missing JavaDoc here, as it is a public constructor.

juancarlovieri commented on src/main/java/commands/ListCommand.java:
```diff
@@ -0,0 +1,38 @@
+public class ListCommand implements Command {
+    private String searchStr = "";
+
+    public ListCommand() {
```
This function might need a JavaDoc since it is public.

juancarlovieri commented on src/main/java/exceptions/LlamaException.java:
```diff
@@ -0,0 +1,10 @@
+ * Represents runtime exceptions that the program might run into
+ */
+public class LlamaException extends RuntimeException {
+    public LlamaException(String message) {
```
I think there is a missing JavaDoc comment here.

juancarlovieri reviewed (commented):
LGTM. Minor styling issues. Love the bot!

bmanara committed:
Move all packages into llama package

bmanara committed:
Implement GUI with echo logic

bmanara committed:
Make GUI to respond similar to CLI

bmanara committed:
Merge branch 'branch-Level-10'

bmanara committed:
Make GUI responsive

bmanara committed:
Improve code quality for Llama class

bmanara committed:
Improve code quality of Ui class

bmanara committed:
Improve exceptions code quality

bmanara committed:
Improve code quality of data

bmanara committed:
Improve code quality of commands

bmanara committed:
Improve code quality of commands

All classes in commands package have improved code quality.
Specifically, avoiding deep nesting and better code structure.

JavaDoc comments have also been ammended to follow convention.

bmanara committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/bmanara/ip into branch-A-CodeQuality

bmanara committed:
Merge pull request #1 from bmanara/branch-A-CodeQuality

Branch a code quality

bmanara committed:
Add assertions to constructors of data

Classes in data package have assertion statements to ensure that it is
filled with appropriate data. Such as decription, endTime and startTime.

bmanara committed:
Merge branch 'master' into branch-A-Assertions

bmanara committed:
Merge pull request #2 from bmanara/branch-A-Assertions

Add assertions to constructors of data

bmanara committed:
Modify build.gradle

bmanara committed:
Add Tag class

bmanara committed:
Add create tag feature

bmanara committed:
Add feature to save and load tags

bmanara committed:
Add feature to tag tasks

bmanara committed:
Add saving tagged tasks feature

bmanara committed:
Add untagging feature

bmanara committed:
Merge branch 'branch-C-Tagging'

bmanara committed:
Change stage title to Llama

bmanara committed:
Add color and speech bubbles

bmanara committed:
Add padding and margin to text

bmanara committed:
Merge branch 'branch-A-BetterGUI'

bmanara committed:
Add GitHub Copilot

bmanara committed:
Update README for User Guide

bmanara committed:
Change user guide notes

bmanara committed:
Add features section

bmanara committed:
Add config.yml file

bmanara committed:
Add Ui screenshot

bmanara committed:
Reformat Table of Contents

bmanara committed:
Add styles.css

bmanara committed:
Remove styles.css

bmanara committed:
Reformat GitHub page

bmanara committed:
Add final touches

bmanara committed:
Merge branch 'branch-A-UserGuide'

bmanara committed:
Fix formatting of GitHub Pages

bmanara committed:
Improve code quality

Improves most JavaDoc comments to follow coding standard

bmanara committed:
Fix bugs regarding tags

bmanara committed:
Edit comment in UntagTaskCommand

bmanara committed:
Update README.md for pages

bmanara committed:
Update README.md
